x-pack/plugins/fleet/server/services/artifacts/mappings.ts

```diff
@@ -7,17 +7,59 @@
 
 import type { ESSearchHit } from '../../../../../../typings/elasticsearch';
 
-import type { Artifact, ArtifactElasticsearchProperties } from './types';
+import type { Artifact, ArtifactElasticsearchProperties, NewArtifact } from './types';
 import { ARTIFACT_DOWNLOAD_RELATIVE_PATH } from './constants';
 
 export const esSearchHitToArtifact = <
   T extends Pick<ESSearchHit<ArtifactElasticsearchProperties>, '_id' | '_source'>
->(
-  searchHit: T
-): Artifact => {
+>({
+  _id: id,
+  _source: {
+    compression_algorithm: compressionAlgorithm,
+    decoded_sha256: decodedSha256,
+    decoded_size: decodedSize,
+    encoded_sha256: encodedSha256,
+    encoded_size: encodedSize,
+    encryption_algorithm: encryptionAlgorithm,
+    package_name: packageName,
+    ...attributesNotNeedingRename
+  },
+}: T): Artifact => {
+  return {
+    ...attributesNotNeedingRename,
+    id,
+    compressionAlgorithm,
+    decodedSha256,
+    decodedSize,
+    encodedSha256,
+    encodedSize,
+    encryptionAlgorithm,
+    packageName,
+  };
+};
+
+export const newArtifactToElasticsearchProperties = ({
+  /* eslint-disable @typescript-eslint/naming-convention */
```

Comment:
I believe you can leave the linting enabled if you leave the destructured properties as camel cased and use snake case when creating/returning the object